libr/core/cmd_api.c

```diff
@@ -813,3 +928,60 @@ R_API char *r_cmd_parsed_args_execstr(RCmdParsedArgs *a) {
 	parsed_args_iterateargs (a, sb);
 	return r_strbuf_drain (sb);
 }
+
+R_API const char *r_cmd_parsed_args_cmd(RCmdParsedArgs *a) {
+	r_return_val_if_fail (a && a->argv && a->argv[0], NULL);
+	return a->argv[0];
+}
+
+/* RCmdDescriptor */
+
+R_API RCmdDesc *r_cmd_desc_argv_new(RCmd *cmd, RCmdDesc *parent, const char *name, RCmdArgvCb cb) {
+	r_return_val_if_fail (cmd && parent && name && cb, NULL);
+	r_return_val_if_fail (parent->type == R_CMD_DESC_TYPE_INNER, NULL);
+	RCmdDesc *res = create_cmd_desc (parent, R_CMD_DESC_TYPE_ARGV, name);
+	if (!res) {
+		return NULL;
+	}
+
+	res->d.argv_data.cb = cb;
+	if (!ht_pp_insert (cmd->ht_cmds, name, res)) {
+		r_cmd_desc_free (res);
+		return NULL;
+	}
+	return res;
+}
+
+R_API RCmdDesc *r_cmd_desc_inner_new(RCmd *cmd, RCmdDesc *parent, const char *name) {
+	r_return_val_if_fail (cmd && parent && name, NULL);
+	r_return_val_if_fail (parent->type == R_CMD_DESC_TYPE_INNER, NULL);
```

Comment:
I am not sure if this condition will still hold once more commands get defined in the tree. Same applies to this condition in  `r_cmd_desc_oldinput_new`.

Reply:
Right now I'd like to keep this invariant, where if a node has children it must be a INNER node. All commands that actually do something should be ARGV/OLDINPUT.